addons/mail/static/src/attachments/attachment_model.js

```diff
@@ -33,6 +33,8 @@ export class Attachment extends Record {
     get isDeletable() {
         if (this.message && this.originThread?.model === "discuss.channel") {
             return this.message.editable;
+        } else if (this.message) {
+            return this.message.allowsEdition;
         }
```

Comment:
@odoo/rd-discuss Hi there, do you think it would be fine here to say "if a message is not editable then its attachments are never deletable? Seems like odd behavior in the first place, unless you had a good reason for it.

I'm emulating the way it works in previous versions here for now. Thanks in advance!

Reply:
Hey! Attachments of a message contribute to all attachments in chatter, accessible through the attachment box button (paper-clip icon). The reason to allow deletion of all attachments in chatter was for this specific case.

There's definitely more improvement to make on the rules to state whether a discuss object is editable or deletable, but right now it's worth keeping the possibility to delete all attachments in chatter.

Reply:
I see, thanks for the insight. I guess this is useful for administrators who might want to clean up records that way, mostly?

Will keep the current behavior of every attachment being deletable, at least for administrators, then.
Cheers!

Reply:
In channel we put permission based on membership / admin of channel.
There's no equivalent concepts in chatter, followers is a looser concept so it doesn't work. For now we assume all users of chatter backend are employees and we assume employees act in good faith ^^

Reply:
Seems reasonable in most cases, especially for incoming attachments. To justify the PR then, idea is that there are circumstances where you can't just trust all employees are acting in good faith, especially if it's messages on a record concerning themselves, and just having access to a document shouldn't mean you can edit important documents sent by someone else.

So only admins should be able to broadly modify or remove attachments, and if you want to modify them for functional reasons you should probably be making a copy.